src/module.c

```diff
@@ -1698,12 +1686,26 @@ int RM_ReplyWithNull(RedisModuleCtx *ctx) {
  * execute some command, as we want to reply to the client exactly the
  * same reply we obtained by the command.
  *
- * The function always returns REDISMODULE_OK. */
+ * Return:
+ * * REDISMODULE_OK On success returns .
+ * * REDISMODULE_ERR if the given reply is in RESP3 format but the client expects RESP2
+ *   In case of error, it's the module writer responsibility to translate the reply
+ *   to RESP2 (or maybe handle it differently by returning an error). Notice that for
+ *   module writer convenient, it is possible to pass `0` as a parameter to the fmt
+ *   argument of `RM_Call` so that the RedisModuleCallReply will return in the same
+ *   format (RESP2 or RESP3) as the current client expects it.
+ * */
```

Comment:
```suggestion
 * - REDISMODULE_ERR if the given reply is in RESP3 format but the client expects RESP2.
 *   In case of an error, it's the module writer responsibility to translate the reply
 *   to RESP2 (or handle it differently by returning an error). Notice that for
 *   module writer convenience, it is possible to pass `0` as a parameter to the fmt
 *   argument of `RM_Call` so that the RedisModuleCallReply will return in the same
 *   protocol (RESP2 or RESP3) as set in the current client's context. */
```